src/App.tsx

```diff
@@ -8,37 +8,144 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Error } from './types/Error';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
+import {
+  deleteComment,
+  getPostComments,
+  getUserPosts,
+  getUsers,
+} from './api/users';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [isLoadingPost, setIsLoadingPost] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [isOpenSidebar, setIsOpenSidebar] = useState(false);
+  const [error, setError] = useState<Error>(Error.None);
+
+  useEffect(() => {
+    getUsers()
+      .then(usersFromData => setUsers(usersFromData));
+  }, []);
+
+  useEffect(() => {
+    setIsLoadingPost(true);
+
+    if (selectedUserId) {
+      setError(Error.None);
+
+      getUserPosts(selectedUserId)
+        .then(posts => setUserPosts(posts))
+        .catch(() => setError(Error.UserPosts))
+        .finally(() => setIsLoadingPost(false));
+    }
+  }, [selectedUserId]);
+
+  useEffect(() => {
+    if (selectedPost) {
+      setIsLoadingComments(true);
+
+      getPostComments(selectedPost.id)
+        .then(comments => setPostComments(comments))
+        .finally(() => setIsLoadingComments(false));
+    }
+  }, [selectedPost]);
+
+  const handleSelectedUser = (userId: number) => {
+    setSelectedUserId(userId);
+    setIsOpenSidebar(false);
+    setError(Error.None);
+
+    setSelectedPost(null);
+  };
+
+  const handleSelectedPost = (postId: number) => {
+    const newSelectedPost = userPosts.find(({ id }) => id === postId);
+
+    if (newSelectedPost) {
+      setSelectedPost(newSelectedPost);
+    }
+
+    setError(Error.None);
+  };
+
+  const handleDeleteComment = (commentId: number) => {
+    setError(Error.None);
+
+    deleteComment(commentId)
+      .then(() => {
+        const newCommentList = postComments
+          .filter(({ id }) => id !== commentId);
+
+        setPostComments(newCommentList);
+      })
+      .catch(() => setError(Error.Delete));
+  };
+
+  const handleToggleSidebar = (status?: boolean) => {
+    if (status) {
+      setIsOpenSidebar(status);
+    } else {
+      setIsOpenSidebar(!isOpenSidebar);
+    }
```

Comment:
Looks a little bit confusing, better to calculate `status` outside this handler, and pass `status` through arguments each time

Reply:
Sorry, but i don't have better solution to work it together ((